### PR TITLE
lib: handle not published packages

### PR DESCRIPTION
**Pull request by BethGriggs**
This PR removes the need for modules to be published to npm before they can be tested. This involved a bit more refactoring than I first anticipated...
 
To summarise, this PR: 
- Introduces `git-url-parse` module
- Extracts the parent module repository from the local `package.json` using the `git-url-parse` module.
  - This enables us to remove `getGitHubOrgRepo`, `getOrgRepo`, and `fetchRegistryInfo` functions.
- Renames variables such as `dep`
  - These changes were initially to just to help me understand the flow of the program, but I think it is worth keeping more specific naming - although happy to change these. 
- Changes the `getPatch` function name to `getCommitURL`
  - Open to bike-shedding on this function name. I felt it should be changed as when I was working through the code I assumed that the function returned a Git Patch, but it returns the URL that we wish to patch the `package.json` with.
- Removes the tests for the removed functions. Updates the test for `getPatch` where the function has been renamed.

### CODE REVIEW

**Comment by BethGriggs** [comment]
It might be easier to review the file in full - https://github.com/pkgjs/wiby/blob/5a124183b53d432170315b541d864a3cdae12a88/lib/test.js

**Comment by BethGriggs** [comment]
Thanks @rodion-arr,  I've addressed the unused var.

I think there's an ongoing discussion about our choice of linter, so I may defer adding standard to the test script for a later PR.

File: lib/test.js [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
+  const parentPkgJSON = await getLocalPackageJSON()
+  const parentPkgInfo = gitURLParse(parentPkgJSON.repository.url)
+  console.log(`Parent module: ${parentPkgInfo.owner}/${parentPkgJSON.name}`)
+
+  const dependentPkgInfo = gitURLParse(url)
+  const dependentPkgJSON = await github.getPackageJson(dependentPkgInfo.owner, dependentPkgInfo.name)
+  console.log(`Dependent module: ${dependentPkgInfo.owner}/${dependentPkgInfo.name}`)
+
+  if (!checkPackageInPackageJSON(parentPkgJSON.name, dependentPkgJSON)) {
     throw new Error('Dependency not found in package.json')
```
Comment (dominykas) [inline lib/test.js:15]: Not a part of this PR, but still:

```suggestion
    throw new Error(`${parentPkgInfo.owner}/${parentPkgJSON.name} not found in package.json of ${dependentPkgInfo.owner}/${dependentPkgInfo.name}`)
```

**Comment by dominykas** [comment]
* [x] At least 1 approval from package-maintenance members other than the author of the PR, unless there is single maintainer for the repository. In that case the single maintainer can land their own PRs.
* [x] No blocking reviews
* [x] 72 hours since the PR was openened

**Pull request merged**